### PR TITLE
pair hooks again

### PR DESCRIPTION
**Pull request by dorin-iancu**

### CODE REVIEW

**Review by alyn509 — APPROVED** [review]

File: dex/pair/src/pair_hooks/call_hook.rs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        }
+
+        let mut output_payments = input_payments;
+        for hook in &hooks {
+            let (_, back_transfers) =
+                ContractCallNoPayment::<_, MultiValueEncoded<ManagedBuffer>>::new(
+                    hook.dest_address,
+                    hook.endpoint_name,
+                )
+                .with_raw_arguments(call_args.clone())
```
Comment (CostinCarabas) [inline dex/pair/src/pair_hooks/call_hook.rs:36]: Is this wanted behavior? The only parameter to be the Pair?
Is there any case where a hook would need personalized parameters?

File: dex/pair/src/pair_hooks/call_hook.rs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                .with_multi_token_transfer(output_payments.clone())
+                .execute_on_dest_context_with_back_transfers::<MultiValueEncoded<ManagedBuffer>>();
+
+            require!(
+                back_transfers.esdt_payments.len() == payments_len,
```
Comment (CostinCarabas) [inline dex/pair/src/pair_hooks/call_hook.rs:41]: The egld value is ignored. Is there any case where EGLD would be used?

File: dex/pair/src/pair_hooks/call_hook.rs (hunk 1 above)
Comment (dorin-iancu) [reply dex/pair/src/pair_hooks/call_hook.rs:36]: Yes, plenty of endpoints have their own parameters.

File: dex/pair/src/pair_hooks/call_hook.rs (hunk 2 above)
Comment (dorin-iancu) [reply dex/pair/src/pair_hooks/call_hook.rs:41]: Nope, only ESDT.

**Review by CostinCarabas — APPROVED** [review]

**Pull request merged**